lib/Service/ExAppPreferenceService.php

```diff
@@ -25,48 +26,73 @@ class ExAppPreferenceService {
 	public function __construct(
 		private ExAppPreferenceMapper $mapper,
 		private LoggerInterface $logger,
+		private ICrypto $crypto,
 	) {
 	}
 
-	public function setUserConfigValue(string $userId, string $appId, string $configKey, mixed $configValue) {
+	public function setUserConfigValue(string $userId, string $appId, string $configKey, mixed $configValue, ?int $sensitive = null): ?ExAppPreference {
 		try {
 			$exAppPreference = $this->mapper->findByUserIdAppIdKey($userId, $appId, $configKey);
 		} catch (DoesNotExistException|MultipleObjectsReturnedException|Exception) {
 			$exAppPreference = null;
 		}
+		if ($sensitive) {
+			try {
+				$encryptedValue = $this->crypto->encrypt($configValue);
+			} catch (\Exception $e) {
+				$this->logger->error('Failed to encrypt sensitive value: ' . $e->getMessage(), ['exception' => $e]);
+				return null;
+			}
+		}
 		if ($exAppPreference === null) {
 			try {
 				return $this->mapper->insert(new ExAppPreference([
 					'userid' => $userId,
 					'appid' => $appId,
 					'configkey' => $configKey,
-					'configvalue' => $configValue ?? '',
+					'configvalue' => $sensitive ? $encryptedValue ?? '' : $configValue ?? '',
+					'sensitive' => $sensitive ?? 0,
 				]));
 			} catch (Exception $e) {
 				$this->logger->error('Error while inserting new config value: ' . $e->getMessage(), ['exception' => $e]);
 				return null;
 			}
 		} else {
-			$exAppPreference->setConfigvalue($configValue);
+			$exAppPreference->setConfigvalue($sensitive ? $encryptedValue : $configValue);
+			if ($sensitive !== null) {
+				$exAppPreference->setSensitive($sensitive);
+			}
 			try {
 				if ($this->mapper->updateUserConfigValue($exAppPreference) !== 1) {
 					$this->logger->error('Error while updating preferences_ex config value');
 					return null;
 				}
-				return $exAppPreference;
 			} catch (Exception $e) {
 				$this->logger->error('Error while updating config value: ' . $e->getMessage(), ['exception' => $e]);
 				return null;
 			}
 		}
+		if ($sensitive) {
+			$exAppPreference->setConfigvalue($configValue);
+		}
```

Comment:
A similar comment for using the plain text value for the sensitive/encrypted field.